konnectivity-client/go.mod

```diff
@@ -5,18 +5,17 @@ go 1.17
 // Prefer to keep requirements compatible with the oldest supported
 // k/k minor version, to prevent client backport issues.
 require (
-	github.com/golang/protobuf v1.4.3
 	go.uber.org/goleak v1.2.0
-	google.golang.org/grpc v1.27.1
+	google.golang.org/grpc v1.48.0
+	google.golang.org/protobuf v1.27.1
```

Comment:
This looks compatible with k/k release-1.24 and 1.23.

Older releases (1.21 and 1.22) use google.golang.org/protobuf v1.26.0. So this change could make backporting there harder. I could imagine making a 0.0.34-backport tag or similar in case we need that.

@tallclair